app/models/ebean/AlertExecution.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2019 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.ebean;
+
+import com.google.common.base.Objects;
+import io.ebean.annotation.DbJsonB;
+import models.internal.AlertEvaluationResult;
+
+import java.time.Instant;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+/**
+ * An execution event for an {@link models.internal.Alert}.
+ * <p>
+ * NOTE: This class is enhanced by Ebean to do things like lazy loading and
+ * resolving relationships between beans. Therefore, including functionality
+ * which serializes the state of the object can be dangerous (e.g. {@code toString},
+ * {@code @Loggable}, etc.).
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+// CHECKSTYLE.OFF: MemberNameCheck
+@Entity
+@Table(name = "alert_executions", schema = "portal")
+@IdClass(AlertExecution.Key.class)
+public final class AlertExecution extends BaseExecution<AlertEvaluationResult> {
+    @Id
+    @ManyToOne(optional = false)
+    @JoinColumn(name = "organization_id")
+    private Organization organization;
+    @Id
+    @Column(name = "alert_id")
+    private UUID alertId;
+    @Nullable
+    @DbJsonB
```

Comment:
This is your likely serialization culprit.

Here's the relevant thread:

https://github.com/ebean-orm/ebean/issues/542

This change related to 542 may offer some help but I have not traced it all the way down:

https://github.com/ebean-orm/ebean/commit/13f7d37e76c0712f7daf59fb689191e4bdba8d8a#diff-2a7fe1eac4aafd91195da026c9a88a9fR675

Specifically, here it looks into `ServerConfig` for an `ObjectMapper`, if we can set that before this method is called to the one we created in Metrics Portal then you would get all the goodies we are used to having with ObjectMapper usage.

If not, the fallback strategy here imo is to not use @DbJsonB and instead perform the serialization/deserialization in the repository object. The only downfall of this imo is that the schema may not be easily upgradable to @DbJsonB in the future (unless you can find a type that works today with manual serde and in the future with Ebean serde).

Reply:
Thanks for the pointers!

I went ahead and configured this in both MainModule and EbeanServerHelper. It looks like it did the trick.

Reply:
Hmm. Apparently it works fine in the unit tests, but I'm actually discarding configuration when trying to use it for integration tests because I don't load the same configuration that play does using the hocon file.

I'm still trying to figure out how to override `ServerConfig` from within Play. If this takes more than an hour I'll probably go with the fallback strategy mentioned.

Reply:
I'm a little confused by what you mean when you say integration tests discard the configuration.

There are two styles of "integration" tests.

1) `*IT.java` code is executed in the Maven JVM against Metrics Portal JVM (e.g. via its APIs) running in Docker. The configuration loaded by Metrics Portal is the one packaged by default into the Docker image.

2) `*IT.java` code is executed in the Maven JVM and loads and runs classes from Metrics Portal against actual dependencies (e.g. Postgresql, Cassandra, KairosDb). There is no configuration in this context in the Maven JVM (nor should there be).

Instead, you will need to have the tests inject the correct `ObjectMapper`. You need to be careful because you want all Ebean tests to inject a single and consistent `ObjectMapper` instance. 

We actually have a similar problem for working with the Play APIs because they use (the same) custom `ObjectMapper`. The code for generating this for tests is here:

https://github.com/ArpNetworking/metrics-portal/blob/master/test/java/com/arpnetworking/testing/SerializationTestUtils.java

You should probably re-use it with some way for Ebean tests (e.g. BaseEbean abstract test class with `@BeforeClass` that does the injection). Shoot me a text and we can talk specifics.

Reply:
These integration tests are the second flavor (only one JVM, no Play APIs, "real" external databases). I understand that there are two kinds but given the context of the PR I thought it was reasonable to omit any specifics.

What I had meant by "discard configuration" is that because of the way this is now set up to configure the `ServerConfig`, this `MetricsPortalServerConfigStartup` class needs to be loaded by ebean in order to call the callback and populate the `ObjectMapper` field. As mentioned below, it has to be called at inject time otherewise the ObjectMapper is null and everything fails.

In the tests, we load the ebean classes manually from within [EbeanServerHelper](https://github.com/ArpNetworking/metrics-portal/blob/2c9237c77c9bd1b05c4b0eb31e9f9b581f6e6407/test/java/com/arpnetworking/metrics/portal/integration/test/EbeanServerHelper.java#L89). This sets the ObjectMapper to something reasonable, but if `MetricsPortalServerConfigStartup` gets loaded here, it will clobber the ObjectMapper. This scenario is what I mean by "discard" - we overwrite any configured value. I was accidentally doing this in one of my earlier commits and it caused my tests to fail.

It sounds like then there are currently 2 (3?) locations where we create an object mapper, and they all vary independently (even though they should all be the same):
* [MainModule](https://github.com/ArpNetworking/metrics-portal/blob/master/app/global/MainModule.java#L290)
* [EbeanServerHelper](https://github.com/ArpNetworking/metrics-portal/blob/2c9237c77c9bd1b05c4b0eb31e9f9b581f6e6407/test/java/com/arpnetworking/metrics/portal/integration/test/EbeanServerHelper.java#L89)
* [SerializationTestUtils](https://github.com/ArpNetworking/metrics-portal/blob/master/test/java/com/arpnetworking/testing/SerializationTestUtils.java)

At the very least I can make the last two the same since they're both run during tests.

Reply:
Yes please! Reducing the two in tests to just one would be very helpful. The duplication and matching with prod is already called out here:

https://github.com/ArpNetworking/metrics-portal/blob/master/test/java/com/arpnetworking/testing/SerializationTestUtils.java#L40

Feel free to file an issue to refactor the Prod code so it can be used from Test as well. That would make it just one definition. Although to be clear, the risk there is someone modifying the prod definition and the tests passing _BUT_ not being backwards compatible. The dual definition should at least catch such configuration changes to ObjectMapper. 

So 2 is better than 3, but 1 may not be better than 2. ;)